### PR TITLE
Issue #1803 Ability to overwrite an image during export

### PR DESCRIPTION
**Pull request by praveenkumar**
How to Test
---------------

```
$ cat Dockerfile 
FROM fedora:latest
MAINTAINER Praveen Kumar "kumarpraveen.nitdgp@gmail.com"

RUN echo "VERSION 0.0.1" >> /etc/os-release

$ docker build -t kumarpraveen/fedora-test:1.0.0 .
Sending build context to Docker daemon 2.048 kB
Step 1/3 : FROM fedora:latest
 ---> cc510acfcd70
Step 2/3 : MAINTAINER Praveen Kumar "kumarpraveen.nitdgp@gmail.com"
 ---> Using cache
 ---> 76e091b492f6
Step 3/3 : RUN echo "VERSION 0.0.1" >> /etc/os-release
 ---> Running in f5bc058b2777
 ---> 44c0270f56ba
Removing intermediate container f5bc058b2777
Successfully built 44c0270f56ba

$ docker push kumarpraveen/fedora-test:1.0.0
The push refers to a repository [docker.io/kumarpraveen/fedora-test]
e551e8575d32: Pushed 
891e1e4ef82a: Layer already exists 
1.0.0: digest: sha256:784299f381169095c668767aa1b1cfe800554e2e1e476c880726a91b05c85d57 size: 736

$ MINISHIFT_ENABLE_EXPERIMENTAL=y ./minishift start --no-provision

$ ./minishift image export kumarpraveen/fedora-test:1.0.0 
Exporting 'kumarpraveen/fedora-test:1.0.0'........... OK
$ ./minishift image list
kumarpraveen/fedora-test:1.0.0

$ python -mjson.tool ~/.minishift/cache/images/index.json 
{
    "manifests": [
        {
            "annotations": {
                "org.opencontainers.image.ref.name": "kumarpraveen/fedora-test:1.0.0"
            },
            "digest": "sha256:41e76a70da76dc810cf3614c7fe7e4430620369bfbc9fae1c5ff473be2971ec1",
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 504
        }
    ],
    "schemaVersion": 2
}

$ cat Dockerfile 
FROM fedora:latest
MAINTAINER Praveen Kumar "kumarpraveen.nitdgp@gmail.com"

RUN echo "VERSION 0.0.2" >> /etc/os-release

$ docker build -t kumarpraveen/fedora-test:1.0.0 .
Sending build context to Docker daemon 2.048 kB
Step 1/3 : FROM fedora:latest
 ---> cc510acfcd70
Step 2/3 : MAINTAINER Praveen Kumar "kumarpraveen.nitdgp@gmail.com"
 ---> Using cache
 ---> 76e091b492f6
Step 3/3 : RUN echo "VERSION 0.0.1" >> /etc/os-release
 ---> Running in f5bc058b2777
 ---> 44c0270f56ba
Removing intermediate container f5bc058b2777
Successfully built 44c0270f56ba

$ docker push kumarpraveen/fedora-test:1.0.0
The push refers to a repository [docker.io/kumarpraveen/fedora-test]
e551e8575d32: Pushed 
891e1e4ef82a: Layer already exists 
1.0.0: digest: sha256:784299f381169095c668767aa1b1cfe800554e2e1e476c880726a91b05c85d57 size: 736

$ ./minishift image export kumarpraveen/fedora-test:1.0.0 --overwrite
Exporting 'kumarpraveen/fedora-test:1.0.0'............. OK

$ python -mjson.tool ~/.minishift/cache/images/index.json 
{
    "manifests": [
        {
            "annotations": {
                "org.opencontainers.image.ref.name": "kumarpraveen/fedora-test:1.0.0"
            },
            "digest": "sha256:9fa243098d333943374444029fa6bd1615a564d4505a5f642ce538d28a915b2c",
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 504
        }
    ],
    "schemaVersion": 2
}

== Also delete the existing VM and start a fresh one ==
$ minishift delete
$ ./minishift start --no-provision
$ ./minishift image import --all
   Importing 'kumarpraveen/fedora-test:1.0.0' ..... OK
$ ./minishift ssh
$ docker run --rm -it  kumarpraveen/fedora-test:1.0.0 cat /etc/os-release
NAME=Fedora
VERSION="28 (Twenty Eight)"
ID=fedora
VERSION_ID=28
PLATFORM_ID="platform:f28"
PRETTY_NAME="Fedora 28 (Twenty Eight)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:28"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=28
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=28
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VERSION 0.0.2  => it have updated one.
```

### CODE REVIEW

File: cmd/minishift/cmd/image/export.go [hunk 1]
```diff
@@ -144,5 +145,6 @@ func createLogFile() *os.File {
 func init() {
 	imageExportCmd.Flags().BoolVar(&exportAll, "all", false, "Exports all images currently available in the Docker daemon.")
 	imageExportCmd.Flags().BoolVar(&logToFile, "log-to-file", false, "Logs export progress to file instead of standard out.")
+	imageExportCmd.Flags().BoolVar(&overwrite, "overwrite", false, "Exports  images currently available in the Docker daemon.")
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/image/export.go]: `--overwrite     Exports  images currently available in the Docker daemon` does not communicate its purpose.  I would suggest `--overwrite     Forces an image exports images  when the image is already available in the Docker daemon`.

**Comment by LalatenduMohanty** [comment]
Tested the PR and it works for me.

File: pkg/minishift/docker/image/oci_image_handler.go [hunk 2]
```diff
@@ -153,8 +153,8 @@ func (handler *OciImageHandler) ExportImages(config *ImageCacheConfig) ([]string
 		progressDots.SetWriter(out)
 		progressDots.Start()
 
-		if !handler.IsImageCached(config, imageName) {
-			err = handler.exportImage(imageName, config, policyContext, out)
+		if !handler.IsImageCached(config, imageName, overwrite) {
```
[the remark this answers is not in the document]
Comment (praveenkumar) [reply pkg/minishift/docker/image/oci_image_handler.go]: @LalatenduMohanty Good catch :+1: , will modify it.

File: cmd/minishift/cmd/image/export.go [hunk 3]
```diff
@@ -144,5 +145,6 @@ func createLogFile() *os.File {
 func init() {
 	imageExportCmd.Flags().BoolVar(&exportAll, "all", false, "Exports all images currently available in the Docker daemon.")
 	imageExportCmd.Flags().BoolVar(&logToFile, "log-to-file", false, "Logs export progress to file instead of standard out.")
+	imageExportCmd.Flags().BoolVar(&overwrite, "overwrite", false, "Forces an image exports when image is already available in the Docker daemon.")
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/image/export.go]: s/exports/export/

**Comment by LalatenduMohanty** [comment]
Merged via https://github.com/minishift/minishift/commit/6353e9f33d3aa79e99ae7d107cb9a716edcf9ef2 . Thanks @praveenkumar

**Pull request closed without merging**